P4/TaskQueries.sql

```diff
@@ -0,0 +1,83 @@
+USE CommercialNetwork
+GO
+
+/*Problem 1*/
+SELECT * FROM Product
+
+
+/*Problem 2*/
+SELECT ProductID, ProductName FROM Product 
+WHERE Price <= 500
+
+
+/*Problem 3*/
+SELECT AVG(Amount) FROM [Order]
+WHERE OrderDate > '01-01-2015' AND OrderDate < '01-31-2015'
+
+
+/*Problem 4*/
+SELECT Product.ProductName, DATEDIFF(YEAR, [Order].OrderDate, GETDATE()) AS Age
+FROM Product INNER JOIN [Order] ON Product.ProductID = [Order].FK_ProductID
+WHERE Product.Price = (SELECT MAX(Price) FROM Product)
+
+
+/*Problem 5*/
+SELECT DISTINCT Product.ProductName
+FROM Product INNER JOIN [Order] ON Product.ProductID = [Order].FK_ProductID
+WHERE [Order].OrderDate > '01-01-2015' AND [Order].OrderDate < '01-31-2015'
+
+
+/*Problem 6*/
```

Comment:
Sum can accept case statements
`select * from 
(
select 
	p.ProductID,
	p.ProductName, 
	SUM(CASE   
		  WHEN YEAR(o.OrderDate) = 2015 and MONTH(o.OrderDate) = 1 THEN o.Quantity  
		  ELSE 0
		END) AS CurrentQuantity,
	SUM(CASE   
		  WHEN YEAR(o.OrderDate) = 2014 and MONTH(o.OrderDate) = 12 THEN o.Quantity  
		  ELSE 0
		END) AS PreviosQuantity
from Product p
inner join [Order] o
on p.ProductID = o.FK_ProductID
where (YEAR(o.OrderDate) = 2015 and MONTH(o.OrderDate) = 1) or (YEAR(o.OrderDate) = 2014 and MONTH(o.OrderDate) = 12)
group by p.ProductID, p.ProductName
) as comaration
where comaration.CurrentQuantity < comaration.PreviosQuantity`